### DISCUSSION TITLE
Programming the Strike Dash, crashes on startup saying "action not set before reading it"

### DISCUSSION THREAD

**Discussion by July2612** [TrianglyRU/OrbinautFramework#29] [Q&A] [answered]
Hey there! 

So I'm currently working on implementing the strike dash ability for my private fork of Orbinaut. I think it's basically finished, but I'm running into a really strange error that's basically forbidding me from testing it. The game directly crashes on startup with the following text:

`___________________________________________
############################################################################################
ERROR in action number 1
of Create Event for object <undefined>:
Variable <unknown_object>.action(100294, -2147483648) not set before reading it.
 at gml_GlobalScript_scr_player_dash (line 69) -        if action != ACTION_STRIKE_DASH && global.dash_type == DASH_CHAOS or global.dash_type == DASH_TTROUBLE
############################################################################################
gml_GlobalScript_scr_player_dash (line 69)
`

If you need to read over my code, I will provide screenshots of it below, including the global variables and macros I defined. I don't think I did anything wrong, since I wrote it similarly to how the rest of the code would do something like this. For the record, line 69 is the very first line of actual code, the one that checks if you're not strike dashing and if your peel out type is the Chaos or Triple Trouble. As for why those macros are there, that's not important. Just know I have a reason for it.

![image](https://github.com/user-attachments/assets/01068102-78a3-47f9-8492-b6750f5a632b)
![image](https://github.com/user-attachments/assets/0f364df6-b159-44c8-a617-80fe288c420a)
![image](https://github.com/user-attachments/assets/22bd9c6b-1b0e-43a9-a6e1-b9e498a2cc66)
![image](https://github.com/user-attachments/assets/15db92cc-e86c-4d6f-9862-b94362da0686)

I inserted this code underneath the default Peel Out code, so that might be causing issues? I could try moving it to its own file. If you need my YYZ file to look over everything, I will provide that as well. 
[The YYZ File](https://www.mediafire.com/file/dbjlxlqtbx7rzav/Orbi2Edit.yyz/file)

If this manages to find you, I would appreciate some advice on how I could fix this. Thank you in advance!

**Comment by TrianglyRU** [comment, marked as answer]
> I inserted this code underneath the default Peel Out code, so that might be causing issues   

The reason this happens is the extra closing bracket at **line 66**. It is closing the function itself

But the script overall is a huge mess the Strike Dash won't probably work correctly at all even if you delete the bracket due to `return false` on the next line

`return` stops the code executing. If `return` is reached, everything below it basically does not exist

I'm breaking down the CD Peelout here but you get the idea. Make a separate script. Pay close attention to the logic of your code
![изображение](https://github.com/user-attachments/assets/92af9577-c3b1-4138-a665-1c8bd310e814)

**Reply by July2612** [reply]
After moving it to its own script, and adding it to different parts of the code alongside where the Peel Out action is used, it works! There is an issue where you can't jump after using it, but I figure that's an easy fix. Thanks a ton!
